tests/Digdir.Domain.Dialogporten.Application.Integration.Tests/Common/DialogApplication.cs

```diff
@@ -55,36 +56,52 @@
             return options;
         });
 
+        _rootProvider = BuildServiceCollection()
+            .BuildServiceProvider();
+
+        await _dbContainer.StartAsync();
+        await EnsureDatabaseAsync();
+        await BuildRespawnState();
+    }
+
+    public void ReplaceSingleton<T>(T instance)
+        where T : class
+    {
+        var serviceCollection = BuildServiceCollection();
+
+        serviceCollection.RemoveAll<T>();
+        serviceCollection.AddSingleton(instance);
+
+        _rootProvider = serviceCollection.BuildServiceProvider();
+    }
```

Comment:
Good point .. 
Also because multiple calls to this within one test would override each other, making it so only one service would be swapped out. 
Gonna refactor 🐰